api/src/main/java/org/infinispan/api/mutiny/MutinyCache.java

```diff
@@ -51,132 +53,366 @@
-   Uni<Boolean> remove(K key);
+   Uni<Boolean> remove(K key, CacheOperationOptions options);
+
+   /**
+    * Removes the key and returns its value if present.
+    *
+    * @param key
+    * @return the value of the key before removal. Returns null if the key didn't exist.
+    */
+   default Uni<V> getAndRemove(K key) {
+      return getAndRemove(key, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Removes the key and returns its value if present.
     *
     * @param key
+    * @param options
     * @return the value of the key before removal. Returns null if the key didn't exist.
     */
-   Uni<V> getAndRemove(K key);
+   Uni<V> getAndRemove(K key, CacheOperationOptions options);
 
    /**
     * Retrieve all keys
     *
     * @return @{@link Multi} which produces keys as items.
     */
-   Multi<K> keys();
+   default Multi<K> keys() {
+      return keys(CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Retrieve all keys
+    *
+    * @return @{@link Multi} which produces keys as items.
+    */
+   Multi<K> keys(CacheOperationOptions options);
+
+   /**
+    * Retrieve all entries
+    *
+    * @return
+    */
+   default Multi<CacheEntry<K, V>> entries() {
+      return entries(CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Retrieve all entries
     *
+    * @param options
+    * @return
+    */
+   Multi<CacheEntry<K, V>> entries(CacheOperationOptions options);
+
+   /**
+    * Retrieve all the entries for the specified keys.
+    *
+    * @param keys
     * @return
     */
-   Multi<? extends CacheEntry<K, V>> entries();
+   default Multi<CacheEntry<K, V>> getAll(Set<K> keys) {
+      return getAll(keys, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Retrieve all the entries for the specified keys.
     *
     * @param keys
+    * @param options
     * @return
     */
-   Multi<? extends CacheEntry<K, V>> getMany(List<K> keys);
+   Multi<CacheEntry<K, V>> getAll(Set<K> keys, CacheOperationOptions options);
 
-   Multi<? extends CacheEntry<K, V>> getMany(K... keys);
+   /**
+    * @param keys
+    * @return
+    */
+   default Multi<CacheEntry<K, V>> getAll(K... keys) {
+      return getAll(CacheOperationOptions.DEFAULT, keys);
+   }
+
+   /**
+    * @param keys
+    * @param options
+    * @return
+    */
+   Multi<CacheEntry<K, V>> getAll(CacheOperationOptions options, K... keys);
 
    /**
     * Put multiple entries from a {@link Multi}
     *
     * @param pairs
     * @return Void
     */
-   Multi<MutinyWriteResult<K>> put(Multi<CacheEntry<K, V>> pairs);
+   default Multi<MutinyWriteResult<K>> put(Multi<CacheEntry<K, V>> pairs) {
+      return put(pairs, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param pairs
+    * @param options
+    * @return
+    */
+   Multi<MutinyWriteResult<K>> put(Multi<CacheEntry<K, V>> pairs, CacheWriteOperationOptions options);
+
+   /**
+    * @param pairs
+    * @return
+    */
+   default Multi<MutinyWriteResult<K>> put(List<CacheEntry<K, V>> pairs) {
```

Comment:
isn't `putAll(Map)` enough like in `AsyncCache`?

Reply:
Yes, I'll remove this

Reply:
Removed